src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http2Connection.cs

```diff
@@ -766,111 +760,184 @@ private void ProcessGoAwayFrame(FrameHeader frameHeader)
+                        {
+                            Debug.Assert(_abortException is not null || _writeQueue.Count > 0, "Write queue signalled but no work to do ??");
+                        }
+#endif
+                    }
+                    else
+                    {
+                        if (writeEntry!.TryDisableCancellation())
+                        {
+                            int writeBytes = writeEntry.WriteBytes;
+
+                            // If the buffer has already grown to 32k, does not have room for the next request,
+                            // and is non-empty, flush the current contents to the wire.
+                            int totalBufferLength = _outgoingBuffer.Capacity;
+                            if (totalBufferLength >= UnflushedOutgoingBufferSize)
+                            {
+                                int activeBufferLength = _outgoingBuffer.ActiveLength;
+                                if (writeBytes >= totalBufferLength - activeBufferLength)
+                                {
+                                    await FlushOutgoingBytesAsync().ConfigureAwait(false);
+                                }
+                            }
+
+                            _outgoingBuffer.EnsureAvailableSpace(writeBytes);
+
+                            try
+                            {
+                                if (NetEventSource.IsEnabled) Trace($"{nameof(writeBytes)}={writeBytes}");
+
+                                // Invoke the callback with the supplied state and the target write buffer.
+                                bool flush = writeEntry.InvokeWriteAction(_outgoingBuffer.AvailableMemorySliced(writeBytes));
+
+                                writeEntry.SetResult();
+
+                                _outgoingBuffer.Commit(writeBytes);
+                                _lastPendingWriterShouldFlush |= flush;
+                            }
+                            catch (Exception e)
+                            {
+                                writeEntry.SetException(e);
+                            }
+                        }
+                    }
+                }
+
+                // Connection is aborting, so fail any outstanding writes.
+                lock (SyncObject)
                 {
-                    _inProgressWrite = FlushOutgoingBytesAsync();
+                    Debug.Assert(_abortException is not null);
+
+                    while (_writeQueue.TryDequeue(out WriteQueueEntry? writeEntry))
+                    {
+                        if (writeEntry!.TryDisableCancellation())
+                        {
+                            writeEntry.SetException(_abortException);
+                        }
+                    }
                 }
             }
+            catch (Exception e)
+            {
+                if (NetEventSource.IsEnabled) Trace($"Unexpected exception in {nameof(ProcessOutgoingFramesAsync)}: {e}");
```

Comment:
You're going to need to rebase on master and change `NetEventSource.IsEnabled` to be `NetEventSource.Log.IsEnabled()` here and elsewhere.  It was modified everywhere in order to better support trimming via the linker.